lib/shipping_agent/deploy.rb

```diff
@@ -39,23 +39,38 @@ def apply
             },
           },
         )
+        Notification.notify("pending", "Config for #{deployment} pushed to kubernetes", url)
       end
+      Thread.new { wait }
+    end
+
+    def wait
+      Timeout.timeout(300) do
+        loop do
+          if deployments.all? { |name| status(name) }
+            Notification.notify("success", "#{app} deployed sucessfully to #{namespace}", url)
+            break
+          end
+          sleep 0.5
+        end
+      end
+    rescue Timeout::Error
+      Notification.notify("error", "#{app} deploy to #{namespace} timed out", url)
     end
 
     private
 
-    def metadata
-      {
-        "labels" => {
-          "version" => version,
-          "build"   => build,
-          "deploy"  => deploy,
-        },
-      }
+    def status(name)
```

Comment:
Can this be renamed to something a bit more descriptive and predicate please?  This is a bit weird, because I wouldn't expect a method called `status` to return a boolean, and `status?` doesn't make sense:

```ruby
50          if deployments.all? { |name| status(name) }
```

How about something like `stable?` or `healthy?`?